### ISSUE TITLE
double check that preprocessing is working in ezcgp

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#125]
it was commented it out Fall2020 in the problem file as some primitives were edited. looks like we have to go in and verify some of the primitives are working. specifically should make sure that the shape of the images never change as that will cause problems down road when building the keras graph.

**Comment by roddtalebi** [comment]
I think i found a problem...
https://augmentor.readthedocs.io/en/master/userguide/extend.html
If we want custom operations, we have to assume that the image will be a PIL image and not np.array. cv2 expects np.array so we have to convert to np.array for all cv2 methods and then convert back to PIL later...but this is kind of familiar...I feel like I played with decorators to do this once upon a time.

**Comment by roddtalebi** [comment]
Also, we use keras_preprocess_func
https://augmentor.readthedocs.io/en/master/_modules/Augmentor/Pipeline.html#Pipeline.keras_preprocess_func

It seems like they expect that the input image is like a np.array with values from 0 to 1 not 0 to 256...is this something keras does or should our data be formated to go from 0 to 1?

**Comment by roddtalebi** [comment]
in this example
https://keras.io/api/preprocessing/image/

when they create fake data, they use np.random.random() so its (0,1) values.

**Comment by roddtalebi** [comment]
i can't find anything ive done on decorators...hopefully it's on my work computer in the 'test_area' folder.  OR I used it in atr project for gtri and im just confusing things.

**Comment by roddtalebi** [comment]
normalize() doesn't seem like it would work with Augmentor.Pipelines...they want to use a list of PIL.Image objects which requires a uint8 data format...normalize means our values will be floats between (0,1)...conflict! going to remove it from operator_dict

**Comment by roddtalebi** [comment]
okay, so if we want to normalize the data, it shouldn't be a primitive but it looks like we can do it in the generator. im going to make this a new issue and leverage everyone else's experience to decide.
https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator
